flat_mutation_reader.hh

```diff
@@ -61,6 +61,11 @@ GCC6_CONCEPT(
     concept bool PartitionFilter = requires(T filter, const dht::decorated_key& dk) {
         { filter(dk) } -> bool;
     };
+
+    template<typename T>
+    concept bool MutationFilter = requires(T filter, const mutation_fragment& mf) {
```

Comment:
`MutationFilter` is already defined in mutation_reader.hh as a different thing.

Reply:
Moved to `flat_mutation_reader.hh` and used by both in next version.